<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ ezGenomeTracks</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ezGenomeTracks</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../get_started.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tracks" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tracks</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tracks"><li><a class="dropdown-item" href="../tracks_coverage.html">Coverage</a></li>
    <li><a class="dropdown-item" href="../tracks_gene.html">Gene</a></li>
    <li><a class="dropdown-item" href="../tracks_sashimi.html">Sashimi</a></li>
    <li><a class="dropdown-item" href="../tracks_manhattan.html">Manhattan</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/Zepeng-Mu/ezGenomeTracks"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="core-functions">Core Functions<a class="anchor" aria-label="anchor" href="#core-functions"></a></h2>

      <div class="section-desc"><p>Main functions for creating genomic tracks</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="ez_coverage.html">ez_coverage()</a></code>

        </dt>
        <dd>Easy coverage track visualization</dd>
      </dl><dl><dt>

          <code><a href="ez_feature.html">ez_feature()</a></code>

        </dt>
        <dd>Easy peak track visualization</dd>
      </dl><dl><dt>

          <code><a href="ez_gene.html">ez_gene()</a></code>

        </dt>
        <dd>Easy gene track visualization</dd>
      </dl><dl><dt>

          <code><a href="ez_link.html">ez_link()</a></code>

        </dt>
        <dd>Easy interaction track visualization</dd>
      </dl><dl><dt>

          <code><a href="ez_sashimi.html">ez_sashimi()</a></code>

        </dt>
        <dd>Easy sashimi plot visualization</dd>
      </dl><dl><dt>

          <code><a href="ez_manhattan.html">ez_manhattan()</a></code>

        </dt>
        <dd>Easy Manhattan plot visualization</dd>
      </dl><dl><dt>

          <code><a href="ez_hic.html">ez_hic()</a></code>

        </dt>
        <dd>Easy Hi-C track visualization</dd>
      </dl><dl><dt>

          <code><a href="genome_plot.html">genome_plot()</a></code>

        </dt>
        <dd>Stack multiple genome tracks</dd>
      </dl></div><div class="section level2">
      <h2 id="geom-functions">Geom Functions<a class="anchor" aria-label="anchor" href="#geom-functions"></a></h2>

      <div class="section-desc"><p>Low-level geometric objects for genomic data</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="geom_coverage.html">geom_coverage()</a></code>

        </dt>
        <dd>Coverage visualization geom</dd>
      </dl><dl><dt>

          <code><a href="geom_feature.html">geom_feature()</a></code>

        </dt>
        <dd>Geom for genomic feature tracks</dd>
      </dl><dl><dt>

          <code><a href="geom_gene.html">geom_gene()</a></code>

        </dt>
        <dd>Geom for gene model tracks</dd>
      </dl><dl><dt>

          <code><a href="geom_link.html">geom_link()</a></code>

        </dt>
        <dd>Geom for genomic interaction links</dd>
      </dl><dl><dt>

          <code><a href="geom_manhattan.html">geom_manhattan()</a></code>

        </dt>
        <dd>Create a Manhattan plot</dd>
      </dl><dl><dt>

          <code><a href="geom_hic.html">geom_hic()</a></code>

        </dt>
        <dd>Geom for Hi-C contact matrix visualization (square heatmap)</dd>
      </dl><dl><dt>

          <code><a href="geom_hic_triangle.html">geom_hic_triangle()</a></code>

        </dt>
        <dd>Geom for Hi-C contact matrix visualization (triangle/rotated view)</dd>
      </dl></div><div class="section level2">
      <h2 id="data-processing-functions">Data Processing Functions<a class="anchor" aria-label="anchor" href="#data-processing-functions"></a></h2>

      <div class="section-desc"><p>Functions for importing and processing track data</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="process_signal_input.html">process_signal_input()</a></code>

        </dt>
        <dd>Process signal input into standardized data frame</dd>
      </dl><dl><dt>

          <code><a href="process_gene_data.html">process_gene_data()</a></code>

        </dt>
        <dd>Process gene annotation data for visualization</dd>
      </dl><dl><dt>

          <code><a href="process_interaction_data.html">process_interaction_data()</a></code>

        </dt>
        <dd>Process interaction data for visualization</dd>
      </dl><dl><dt>

          <code><a href="process_sashimi_data.html">process_sashimi_data()</a></code>

        </dt>
        <dd>Process sashimi plot data for visualization</dd>
      </dl><dl><dt>

          <code><a href="process_manhattan_input.html">process_manhattan_input()</a></code>

        </dt>
        <dd>Process Manhattan plot input into standardized data frame</dd>
      </dl><dl><dt>

          <code><a href="process_hic_data.html">process_hic_data()</a></code>

        </dt>
        <dd>Process Hi-C data for visualization</dd>
      </dl><dl><dt>

          <code><a href="import_genomic_data.html">import_genomic_data()</a></code>

        </dt>
        <dd>Import genomic data from a file into a tidy data frame</dd>
      </dl><dl><dt>

          <code><a href="interaction_track.html">interaction_track()</a></code>

        </dt>
        <dd>Create an interaction track from a file</dd>
      </dl></div><div class="section level2">
      <h2 id="themes-and-scales">Themes and Scales<a class="anchor" aria-label="anchor" href="#themes-and-scales"></a></h2>

      <div class="section-desc"><p>Functions for customizing track appearance</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="ez_theme.html">ez_theme()</a></code>

        </dt>
        <dd>A minimal theme for genome track visualization</dd>
      </dl><dl><dt>

          <code><a href="ez_coverage_theme.html">ez_coverage_theme()</a></code>

        </dt>
        <dd>A theme for coverage tracks</dd>
      </dl><dl><dt>

          <code><a href="ez_feature_theme.html">ez_feature_theme()</a></code>

        </dt>
        <dd>A theme for feature tracks</dd>
      </dl><dl><dt>

          <code><a href="ez_gene_theme.html">ez_gene_theme()</a></code>

        </dt>
        <dd>A theme for gene tracks</dd>
      </dl><dl><dt>

          <code><a href="ez_sashimi_theme.html">ez_sashimi_theme()</a></code>

        </dt>
        <dd>A theme for sashimi plots</dd>
      </dl><dl><dt>

          <code><a href="ez_manhattan_theme.html">ez_manhattan_theme()</a></code>

        </dt>
        <dd>A theme for Manhattan plots</dd>
      </dl><dl><dt>

          <code><a href="ez_hic_theme.html">ez_hic_theme()</a></code>

        </dt>
        <dd>Hi-C theme</dd>
      </dl><dl><dt>

          <code><a href="scale_x_genome.html">scale_x_genome()</a></code>

        </dt>
        <dd>Scale for genomic coordinates on the x-axis</dd>
      </dl><dl><dt>

          <code><a href="scale_x_genome_region.html">scale_x_genome_region()</a></code>

        </dt>
        <dd>Create a genomic position scale for a specific chromosome region</dd>
      </dl><dl><dt>

          <code><a href="scale_y_strand.html">scale_y_strand()</a></code>

        </dt>
        <dd>Y-axis scale for strand tracks</dd>
      </dl><dl><dt>

          <code><a href="scale_fill_hic.html">scale_fill_hic()</a></code>

        </dt>
        <dd>Scale for Hi-C contact matrix fill colors</dd>
      </dl><dl><dt>

          <code><a href="hic_palettes.html">hic_palette_cooler()</a></code> <code><a href="hic_palettes.html">hic_palette_ylgnbu()</a></code> <code><a href="hic_palettes.html">hic_palette_viridis()</a></code> <code><a href="hic_palettes.html">hic_palette_bwr()</a></code>

        </dt>
        <dd>Hi-C color palettes</dd>
      </dl></div><div class="section level2">
      <h2 id="annotation-functions">Annotation Functions<a class="anchor" aria-label="anchor" href="#annotation-functions"></a></h2>

      <div class="section-desc"><p>Functions for adding annotations to tracks</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="add_vline.html">add_vline()</a></code>

        </dt>
        <dd>Add a vertical line to a genome track</dd>
      </dl><dl><dt>

          <code><a href="add_hline.html">add_hline()</a></code>

        </dt>
        <dd>Add a horizontal line to a genome track</dd>
      </dl><dl><dt>

          <code><a href="add_rect.html">add_rect()</a></code>

        </dt>
        <dd>Add a rectangle to a genome track</dd>
      </dl><dl><dt>

          <code><a href="add_text.html">add_text()</a></code>

        </dt>
        <dd>Add text annotation to a genome track</dd>
      </dl><dl><dt>

          <code><a href="add_highlight.html">add_highlight()</a></code>

        </dt>
        <dd>Add highlight regions to a genome track</dd>
      </dl></div><div class="section level2">
      <h2 id="helper-functions">Helper Functions<a class="anchor" aria-label="anchor" href="#helper-functions"></a></h2>

      <div class="section-desc"><p>Utility functions for data conversion and calculation</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="granges_to_df.html">granges_to_df()</a></code>

        </dt>
        <dd>Convert a GRanges object to a tidy data frame</dd>
      </dl><dl><dt>

          <code><a href="df_to_granges.html">df_to_granges()</a></code>

        </dt>
        <dd>Convert a data frame to a GRanges object</dd>
      </dl><dl><dt>

          <code><a href="parse_region.html">parse_region()</a></code>

        </dt>
        <dd>Parse a genomic region string into a GRanges object</dd>
      </dl><dl><dt>

          <code><a href="format_genomic_coord.html">format_genomic_coord()</a></code>

        </dt>
        <dd>Format genomic coordinates</dd>
      </dl><dl><dt>

          <code><a href="extract_txdb_data.html">extract_txdb_data()</a></code>

        </dt>
        <dd>Extract gene data from TxDb object</dd>
      </dl><dl><dt>

          <code><a href="get_single_signal.html">get_single_signal()</a></code>

        </dt>
        <dd>Extract signal data for a single input element</dd>
      </dl><dl><dt>

          <code><a href="plot_signal_df.html">plot_signal_df()</a></code>

        </dt>
        <dd>Plot signal tracks from a data frame with grouping</dd>
      </dl><dl><dt>

          <code><a href="calculate_link_ylim.html">calculate_link_ylim()</a></code>

        </dt>
        <dd>Calculate y-axis limits for link tracks</dd>
      </dl><dl><dt>

          <code><a href="stat_bin_signal.html">stat_bin_signal()</a></code>

        </dt>
        <dd>Stat for binning genomic signal data</dd>
      </dl></div><div class="section level2">
      <h2 id="example-datasets">Example Datasets<a class="anchor" aria-label="anchor" href="#example-datasets"></a></h2>

      <div class="section-desc"><p>Example datasets for demonstration</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="example_signal.html">example_signal</a></code>

        </dt>
        <dd>Example signal data</dd>
      </dl><dl><dt>

          <code><a href="example_peaks.html">example_peaks</a></code>

        </dt>
        <dd>Example peak data</dd>
      </dl><dl><dt>

          <code><a href="example_genes.html">example_genes</a></code>

        </dt>
        <dd>Example gene data</dd>
      </dl><dl><dt>

          <code><a href="example_interactions.html">example_interactions</a></code>

        </dt>
        <dd>Example interaction data</dd>
      </dl><dl><dt>

          <code><a href="example_junctions.html">example_junctions</a></code>

        </dt>
        <dd>Example splice junction data for sashimi plots</dd>
      </dl><dl><dt>

          <code><a href="example_hic.html">example_hic</a></code>

        </dt>
        <dd>Example Hi-C data</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zepeng Mu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

